src/circuit/gadget/utilities/lookup_range_check.rs

```diff
@@ -342,8 +337,14 @@ impl<F: FieldExt + PrimeFieldBits, const K: usize> LookupRangeCheckConfig<F, K>
             || shifted.ok_or(Error::SynthesisError),
         )?;
 
-        // Enable lookup for shifted element, to constrain it to 10 bits.
-        self.q_lookup_short.enable(region, 1)?;
+        // Assign 2^{-num_bits} from a fixed column.
+        let inv_two_pow_s = F::from_u64(1 << num_bits).invert().unwrap();
+        region.assign_advice_from_constant(
+            || format!("2^(-{})", num_bits),
+            self.running_sum,
+            2,
+            inv_two_pow_s,
+        )?;
```

Comment:
This makes this region 50% larger (3 rows instead of 2), but I think this is an acceptable trade-off here:
- It changes the gate from using rotations `-1,0` to using `-1,0,1` which other gates already use, so we shouldn't be gaining proof size from it.
- If we instead split the combined selector into a selector and a fixed column, we would need to keep that fixed column around (costing 32 bytes in the proof), because we want these to be layed out next to the fixed-base scalar mul regions (so the Lagrange coefficient columns are occupied).

![image](https://user-images.githubusercontent.com/4993799/126843299-0a13046d-cf7b-40b3-8182-da93c18b4f9a.png)

Reply:
We could alternatively use 2 columns to reduce this back to 2 rows/rotations, and keep more 4-column space free:

![image](https://user-images.githubusercontent.com/4993799/126848563-8ce90192-b324-4a25-912d-d957d6ea4861.png)

Reply:
I guess we don't yet know if 2-, 3-, or 4-column space would be more useful for future applications. For now, I'm in favour of keeping it to a single column.